hipparchus-core/src/main/java/org/hipparchus/analysis/differentiation/Derivative.java

```diff
@@ -64,4 +71,64 @@ public interface Derivative<T extends CalculusFieldElement<T>> extends CalculusF
     T compose(double... f)
         throws MathIllegalArgumentException;
 
+    /** {@inheritDoc} */
+    @Override
+    default T log10() {
+        return log().divide(FastMath.log(10.));
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    default T pow(T e) {
+        return log().multiply(e).exp();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    default T cosh() {
+        return (exp().add(negate().exp())).divide(2);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    default T sinh() {
+        return (exp().subtract(negate().exp())).divide(2);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    default T acos() {
+        return asin().negate().add(getPi().divide(2));
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    default T ceil() {
+        return newInstance(FastMath.ceil(getValue()));
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    default T floor() {
+        return newInstance(FastMath.floor(getValue()));
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    default T rint() {
+        return newInstance(FastMath.rint(getValue()));
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    default T ulp() {
+        return newInstance(FastMath.ulp(getValue()));
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    default T sign() {
+        return newInstance(FastMath.signum(getValue()));
+    }
+
```

Comment:
Just a question: why are these default implementations in Derivative and not CalculuFieldElement?

Reply:
It's because there is no getValue() in CalculusFieldElement, only getReal() and I thought that using it there might be too strong of a default assumption.

Reply:
Well, these methods are really linked to the duality with real, so I guess they could be pushed upwards in the inheritance tree.

Reply:
Ok, I'll move it upward, but will still overwrite it in FieldDerivative using getValue with as the latter is not a double in that case and is not equivalent to getReal

Reply:
Actually I think I'll leave it as is for this PR and deal with it when talking issue #286